daemons/execd/execd_commands.c

```diff
@@ -507,6 +507,17 @@ send_client_notify(gpointer key, gpointer value, gpointer user_data)
         crm_trace("Skipping notification to client without name");
         return;
     }
+    if (pcmk__starts_with(client->name, "proxy-")) {
+        /* We only want to notify executor IPC clients. If we are running as
+         * Pacemaker Remote, we may have clients proxied to other IPC services
+         * running on the cluster node hosting our connection (identified by
+         * ipc_proxy_accept() with client names starting with "proxy-", which
+         * may not be a perfect way of detecting them, but we don't store the
+         * server IPC name in the client object).
+         */
+        crm_trace("Skipping notification to %s IPC client", client->name);
+        return;
+    }
 
```

Comment:
looks ok I guess.
if we want to avoid relying on client-names and having to do string-checking there would still be the client-flags.
pcmk__client_proxied is for the part on the cluster-node if I get it right but we could add an additional flag.

Reply:
I think a flag does make sense. If I follow correctly, it goes like this:

* The client on the remote node connects to the local IPC name, which is actually pacemaker-remoted listening on all names. ipc_proxy_accept() accepts the connection and creates the client object including the "proxy-" name. I think this is where we should set the flag.

* ipc_proxy_dispatch() in pacemaker-remoted handles all traffic received on that connection from the client, and sets crm_ipc_proxied when it forwards the request to the controller hosting the remote connection.

* remote_proxy_cb() in the controller grabs the flags (including crm_ipc_proxied) from the request XML and passes them when forwarding the original IPC request to the actual IPC server. (Controller API requests are an exception; the controller handles those directly without calling remote_proxy_cb().)

* When an IPC server receives a request, pcmk__client_data2xml() sets pcmk__client_proxied if crm_ipc_proxied is set. This is used only to set crm_ipc_proxied_relay_response on any response sent back to the client, which in turn is used only to decide whether to set F_LRMD_IPC_OP to LRMD_IPC_OP_EVENT or LRMD_IPC_OP_RESPONSE when sending data back to the client on the remote node.

I'll redo this with a flag.